### PR TITLE
AM-688: Do not refer to company.com

### PR DESCRIPTION
**Pull request by mstasgravitee**
[AM-688](https://gravitee.atlassian.net/browse/AM-688)

[AM-688]: https://gravitee.atlassian.net/browse/AM-688?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java
Comment (mstasgravitee) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java:120]: I do not introduce anything new. The previous value was with auth: "https://auth.company.com".

File: gravitee-am-ui/src/app/domain/applications/application/overview/overview.component.html
Comment (mstasgravitee) [reply gravitee-am-ui/src/app/domain/applications/application/overview/overview.component.html:334]: it is the same as in 356 line.

File: gravitee-am-ui/src/app/domain/applications/application/overview/overview.component.html
Comment (mstasgravitee) [reply gravitee-am-ui/src/app/domain/applications/application/overview/overview.component.html:231]: it is the same as in 356 line.

Comment (leleueri) [reply gravitee-am-ui/src/app/domain/applications/application/overview/overview.component.html:231]: see https://github.com/gravitee-io/gravitee-access-management/pull/3275#discussion_r1420605548

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/EmailServiceImpl.java
Comment (leleueri) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/EmailServiceImpl.java:264]: **todo**: For a URL `http://www.gravitee.io/path`, `URI.getHost()` will provide  `www.gravitee.io`.
So if I understand the intension (provide a vairable in the email template engine to make the noreply adapted to the domain), maybe the following change is more appropriate.

```suggestion
        params.put("gatewayHost", domainService.gatewayUrl().getHost().replaceFirst("^www\\.", ""));
```

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java:228]: **issue**: I don't think it is a good think to create a new instance of the DomainService here as this service is a Spring Component (instantiate as SIngleton), it maybe better to provide tthe service (maybe with Lazy ?) but by doing so, should we introduce cyclic dependency into the bean injection :thinking:

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java
Comment (leleueri) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java:70]: **Thought**: Injection DomainService into EmailService looks stranve to me.

**Review by leleueri — CHANGES_REQUESTED** [review]
I'm wondering if instead of injecting DomainService into EmailService & EntrypointService it should be interesting to introduce a "GatewayConfiguration" (not sure about the name) that will read the `gateway.url` value and provide method to get only the host and another method to get the value as defined into the gravitee.yaml (with validation of the URL format into the constructor).

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java
Comment (mstasgravitee) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java:70]: In EntrypointServiceImpl it has to be injected

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/email/EmailServiceTest.java
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/email/EmailServiceTest.java:38]: **todo**: useless import

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/spring/CommonConfiguration.java:61]: **todo**: useless import ?

**Review by leleueri — COMMENTED** [review]
LGTM just two minor comment about useless import.
Need to do some tests before merging, so I will not approve for now. (WIll try to do it today)

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java
Comment (leleueri) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/EntrypointServiceImpl.java:70]: ok

File: gravitee-am-policy/gravitee-am-policy-send-email/src/main/resources/schemas/schema-form.json
Comment (leleueri) [inline gravitee-am-policy/gravitee-am-policy-send-email/src/main/resources/schemas/schema-form.json:13]: **issue**: it was a good idea but it doesn't work. We introduced a email format check and the domain must match a value defined into the gavitee.yaml. So it is not possible to use a template variable for the domain. If you want to test, you can try to save a flow with the SendEmail policy. 

image

Don't know if we have to remove the template variable here or if we have to add a matcher for this specific variable. :thinking: Maybe we should replace this template variable by `gravitee.io` to avoid potential issue in the validation (managed by `io.gravitee.am.service.validators.email.EmailDomainValidatorImpl`)

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/EmailServiceImpl.java
Comment (leleueri) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/EmailServiceImpl.java:264]: **question**: As it is not possible to use variable in the template for the "from" header, should we keep this variable :thinking: ? See https://github.com/gravitee-io/gravitee-access-management/pull/3275/files#diff-18fe570831b203224bdfd7400439c0778b685600a6ac699c057031f89c5c20afR13

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java:255]: **question**: As it is not possible to use variable in the template for the "from" header, should we keep this variable :thinking: ? see https://github.com/gravitee-io/gravitee-access-management/pull/3275/files#diff-18fe570831b203224bdfd7400439c0778b685600a6ac699c057031f89c5c20afR13

**Review by leleueri — CHANGES_REQUESTED** [review]

**Review by leleueri — APPROVED** [review]

**Pull request merged**
